sdjournal/journal.go

```diff
@@ -291,6 +291,19 @@ func (j *Journal) Wait(timeout time.Duration) int {
 	return int(r)
 }
 
+// WaitIndefinitely will synchronously wait until the journal gets changed.  This
+// call will sleep indefinitely while it waits for a change.
+func (j *Journal) WaitIndefinitely() int {
+        j.mu.Lock()
+        // sd_journal_wait(3) calls for a (uint64_t) -1 to be passed to signify
+        // indefinite wait, but using a -1 overflows our C.uint64_t, so we use an
+        // equivalent hex value
+        r := C.sd_journal_wait(j.cjournal, (C.uint64_t)(0xffffffffffffffff))
```

Comment:
Alternatively we could just do this in CGo:
```
/*
int my_journal_wait_indefinitely(sd_journal* j) {
        return sd_journal_wait(j, (uint64_t) -1);
}
*/
import "C"

...

r := C.my_journal_wait_indefinitely(j.cjournal)
```

Not sure if there's a reason to prefer one way or another - @vcaputo ?